app/controllers/orders_controller.rb

```diff
@@ -326,7 +314,7 @@ def index
   # Group into chunks by the group id. If group_id is nil, then it should get its
   # own chunk.
   def grouped_order_details
-    @order.order_details.order(:group_id).slice_when do |before, after|
+    @order.order_details.sort_by(&:group_id).slice_when do |before, after|
```

Comment:
Never mind, this is necessary. I forgot why I did this. You are correct that it takes it out of AR and into memory. But we are doing that anyways by `slice_when`. When done this way, `@order.order_details` will be memoized on `@order`. With the `order`, it returns a fresh set of objects (even though they have the same values). We run the validations on `@order.order_details`, but render the objects returned from `grouped_order_detail`. I want to render the same objects that get the validations (and therefore have `errors` on them) in order to display the validation errors in line and in red.

I'll add a comment explaining that since this is definitely non-obvious, and even I forgot why I made the change.